src/app/component/CurrencySwitcher/CurrencySwitcher.component.js

```diff
@@ -31,15 +31,28 @@ export class CurrencySwitcher extends PureComponent {
         handleCurrencySelect: PropTypes.func.isRequired
     };
 
-    render() {
+    valueForSwitcher() {
         const {
-            handleCurrencySelect,
             currencyData: {
                 available_currencies_data: availableCurrencies,
                 current_currency_code: currentCurrencyCode
             } = {}
         } = this.props;
 
+        const currency = getCurrency();
+        const value = availableCurrencies.some((e) => e.id === currency) ? currency : currentCurrencyCode;
+
+        return value;
+    }
```

Comment:
Please comment on this, it is unclear how this should function and why so.

Reply:
Save the value of the selected currency to the currency variable. Then  iterate over all the available currencies and if the selected currency matches the available ones, then we return the selected one, if not, then we return the default currency

Reply:
When I say that the logic is not obvious and ask you to comment on it, I imply that the next person who is going to work with this code (e.g. me) will not understand it in 10 seconds, and that means that this piece of code should be commented on directly in the code. Nobody will look for the information in the CRs :)